### PR TITLE
HDDS-976: Parse network topology from yaml file

### PR DESCRIPTION
**Pull request by cjjnjust**
This is a early stage patch to support load HDFS schema from a YAML file. 

It should need more unit tests.

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 39 | Docker mode activated. |
||| _ Prechecks _ |
| 0 | yamllint | 0 | yamllint was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 4 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1237 | trunk passed |
| +1 | compile | 59 | trunk passed |
| +1 | checkstyle | 19 | trunk passed |
| +1 | mvnsite | 40 | trunk passed |
| +1 | shadedclient | 774 | branch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 74 | trunk passed |
| +1 | javadoc | 40 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 51 | the patch passed |
| +1 | compile | 35 | the patch passed |
| +1 | javac | 35 | the patch passed |
| -0 | checkstyle | 16 | hadoop-hdds/common: The patch generated 13 new + 0 unchanged - 0 fixed = 13 total (was 0) |
| +1 | mvnsite | 42 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 1 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 820 | patch has no errors when building and testing our client artifacts. |
| -1 | findbugs | 79 | hadoop-hdds/common generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | javadoc | 35 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 79 | common in the patch failed. |
| +1 | asflicense | 25 | The patch does not generate ASF License warnings. |
| | | 3547 | |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:hadoop-hdds/common |
|  |  org.apache.hadoop.hdds.scm.net.NodeSchemaLoader.loadSchemaFromYaml(File) may fail to clean up java.io.InputStream  Obligation to clean up resource created at NodeSchemaLoader.java:up java.io.InputStream  Obligation to clean up resource created at NodeSchemaLoader.java:[line 208] is not discharged |
| Failed junit tests | hadoop.hdds.scm.net.TestNetworkTopologyImpl |
|   | hadoop.hdds.scm.net.TestNodeSchemaManager |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-661/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/661 |
| JIRA Issue | HDDS-976 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  xml  findbugs  checkstyle  yamllint  |
| uname | Linux 6a50ad832710 4.4.0-138-generic #164~14.04.1-Ubuntu SMP Fri Oct 5 08:56:16 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / f41f938 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/1/artifact/out/diff-checkstyle-hadoop-hdds_common.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/1/artifact/out/new-findbugs-hadoop-hdds_common.html |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/1/artifact/out/patch-unit-hadoop-hdds_common.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/1/testReport/ |
| Max. process+thread count | 304 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common U: hadoop-hdds/common |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 42 | Docker mode activated. |
||| _ Prechecks _ |
| 0 | yamllint | 0 | yamllint was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 4 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1144 | trunk passed |
| +1 | compile | 48 | trunk passed |
| +1 | checkstyle | 21 | trunk passed |
| +1 | mvnsite | 39 | trunk passed |
| +1 | shadedclient | 767 | branch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 74 | trunk passed |
| +1 | javadoc | 41 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 44 | the patch passed |
| +1 | compile | 32 | the patch passed |
| +1 | javac | 32 | the patch passed |
| -0 | checkstyle | 15 | hadoop-hdds/common: The patch generated 12 new + 0 unchanged - 0 fixed = 12 total (was 0) |
| +1 | mvnsite | 36 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 1 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 782 | patch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 76 | the patch passed |
| +1 | javadoc | 36 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 75 | common in the patch failed. |
| +1 | asflicense | 26 | The patch does not generate ASF License warnings. |
| | | 3382 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.net.TestNetworkTopologyImpl |
|   | hadoop.hdds.scm.net.TestNodeSchemaManager |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-661/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/661 |
| JIRA Issue | HDDS-976 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  xml  findbugs  checkstyle  yamllint  |
| uname | Linux 143f4e0000dd 4.4.0-138-generic #164~14.04.1-Ubuntu SMP Fri Oct 5 08:56:16 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 82769ba |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/2/artifact/out/diff-checkstyle-hadoop-hdds_common.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/2/artifact/out/patch-unit-hadoop-hdds_common.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/2/testReport/ |
| Max. process+thread count | 340 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common U: hadoop-hdds/common |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NodeSchemaLoader.java [hunk 1]
```diff
@@ -165,6 +169,81 @@ private NodeSchemaLoadResult loadSchema(File schemaFile) throws
...
+   * Load network topology layer schemas from a YAML configuration file.
+   * @param schemaFile schema file
+   * @return all valid node schemas defined in schema file
+   * @throws ParserConfigurationException ParserConfigurationException happen
+   * @throws IOException no such schema file
+   * @throws SAXException xml file has some invalid elements
+   * @throws IllegalArgumentException xml file content is logically invalid
+   */
+  private NodeSchemaLoadResult loadSchemaFromYaml(File schemaFile) {
+    LOG.info("Loading network topology layer schema file " + schemaFile);
```
Comment (xiaoyuyao) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NodeSchemaLoader.java]: NIT:  can we use parameterized log4j like below
LOG.info("Loading network topology layer schema file {}", schemaFile);

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NodeSchemaLoader.java [hunk 2]
```diff
@@ -165,6 +169,81 @@ private NodeSchemaLoadResult loadSchema(File schemaFile) throws
...
+    NodeSchemaLoadResult finalSchema;
+
+    try {
+      FileInputStream fileInputStream = new FileInputStream(schemaFile);
```
Comment (xiaoyuyao) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NodeSchemaLoader.java]: Can we use try-with-resource to ensure the FileInputStream is closed properly?

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NodeSchemaManager.java [hunk 3]
```diff
@@ -59,13 +59,20 @@ public void init(Configuration conf) {
...
+            ScmConfigKeys.OZONE_SCM_NETWORK_TOPOLOGY_SCHEMA_FILE_TYPE);
+
     String schemaFile = conf.get(
         ScmConfigKeys.OZONE_SCM_NETWORK_TOPOLOGY_SCHEMA_FILE,
         ScmConfigKeys.OZONE_SCM_NETWORK_TOPOLOGY_SCHEMA_FILE_DEFAULT);
 
     NodeSchemaLoadResult result;
     try {
-      result = NodeSchemaLoader.getInstance().loadSchemaFromFile(schemaFile);
+      if (schemaFileType.compareTo("yaml") == 0) {
```
Comment (xiaoyuyao) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NodeSchemaManager.java]: Can we make the schema type string case insensitive?

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NodeSchemaManager.java [hunk 4]
```diff
@@ -59,13 +59,20 @@ public void init(Configuration conf) {
...
+        result = NodeSchemaLoader.getInstance().loadSchemaFromYaml(schemaFile);
+      } else {
+        result = NodeSchemaLoader.getInstance().loadSchemaFromFile(schemaFile);
```
Comment (xiaoyuyao) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NodeSchemaManager.java]: Maybe change this to loadSchemaFromXml?

**Comment by xiaoyuyao** [comment]
Thanks @cjjnjust  for working on this. Patch LGTM overall, just few minor issues commented inline.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 28 | Docker mode activated. |
||| _ Prechecks _ |
| 0 | yamllint | 0 | yamllint was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 5 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1096 | trunk passed |
| +1 | compile | 37 | trunk passed |
| +1 | checkstyle | 17 | trunk passed |
| +1 | mvnsite | 37 | trunk passed |
| +1 | shadedclient | 696 | branch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 71 | trunk passed |
| +1 | javadoc | 35 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 43 | the patch passed |
| +1 | compile | 34 | the patch passed |
| +1 | javac | 34 | the patch passed |
| -0 | checkstyle | 14 | hadoop-hdds/common: The patch generated 6 new + 0 unchanged - 0 fixed = 6 total (was 0) |
| +1 | mvnsite | 37 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 1 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 732 | patch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 77 | the patch passed |
| +1 | javadoc | 33 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 79 | common in the patch failed. |
| +1 | asflicense | 23 | The patch does not generate ASF License warnings. |
| | | 3169 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.net.TestNodeSchemaManager |
|   | hadoop.hdds.scm.net.TestNetworkTopologyImpl |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-661/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/661 |
| JIRA Issue | HDDS-976 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  xml  findbugs  checkstyle  yamllint  |
| uname | Linux a3473c677382 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / a92806d |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/3/artifact/out/diff-checkstyle-hadoop-hdds_common.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/3/artifact/out/patch-unit-hadoop-hdds_common.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/3/testReport/ |
| Max. process+thread count | 422 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common U: hadoop-hdds/common |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 27 | Docker mode activated. |
||| _ Prechecks _ |
| 0 | yamllint | 0 | yamllint was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 5 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1141 | trunk passed |
| +1 | compile | 50 | trunk passed |
| +1 | checkstyle | 19 | trunk passed |
| +1 | mvnsite | 40 | trunk passed |
| +1 | shadedclient | 699 | branch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 75 | trunk passed |
| +1 | javadoc | 36 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 43 | the patch passed |
| +1 | compile | 31 | the patch passed |
| +1 | javac | 31 | the patch passed |
| -0 | checkstyle | 14 | hadoop-hdds/common: The patch generated 6 new + 0 unchanged - 0 fixed = 6 total (was 0) |
| +1 | mvnsite | 36 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 1 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 729 | patch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 82 | the patch passed |
| +1 | javadoc | 34 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 67 | common in the patch failed. |
| +1 | asflicense | 24 | The patch does not generate ASF License warnings. |
| | | 3221 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.net.TestNodeSchemaManager |
|   | hadoop.hdds.scm.net.TestNetworkTopologyImpl |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-661/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/661 |
| JIRA Issue | HDDS-976 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  xml  findbugs  checkstyle  yamllint  |
| uname | Linux 40077f97363b 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / d2637cb |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/4/artifact/out/diff-checkstyle-hadoop-hdds_common.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/4/artifact/out/patch-unit-hadoop-hdds_common.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/4/testReport/ |
| Max. process+thread count | 446 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common U: hadoop-hdds/common |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/4/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 27 | Docker mode activated. |
||| _ Prechecks _ |
| 0 | yamllint | 0 | yamllint was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 5 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 24 | Maven dependency ordering for branch |
| +1 | mvninstall | 1031 | trunk passed |
| +1 | compile | 76 | trunk passed |
| +1 | checkstyle | 33 | trunk passed |
| +1 | mvnsite | 76 | trunk passed |
| +1 | shadedclient | 749 | branch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 120 | trunk passed |
| +1 | javadoc | 69 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 12 | Maven dependency ordering for patch |
| -1 | mvninstall | 21 | container-service in the patch failed. |
| -1 | compile | 51 | hadoop-hdds in the patch failed. |
| -1 | javac | 51 | hadoop-hdds in the patch failed. |
| +1 | checkstyle | 22 | the patch passed |
| -1 | mvnsite | 24 | container-service in the patch failed. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 2 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 774 | patch has no errors when building and testing our client artifacts. |
| -1 | findbugs | 23 | container-service in the patch failed. |
| +1 | javadoc | 57 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 63 | common in the patch failed. |
| -1 | unit | 24 | container-service in the patch failed. |
| +1 | asflicense | 25 | The patch does not generate ASF License warnings. |
| | | 3500 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.net.TestNodeSchemaManager |
|   | hadoop.hdds.scm.net.TestNetworkTopologyImpl |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-661/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/661 |
| JIRA Issue | HDDS-976 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  xml  findbugs  checkstyle  yamllint  |
| uname | Linux bf4f15e5091e 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / d2637cb |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/5/artifact/out/patch-mvninstall-hadoop-hdds_container-service.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/5/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/5/artifact/out/patch-compile-hadoop-hdds.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/5/artifact/out/patch-mvnsite-hadoop-hdds_container-service.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/5/artifact/out/patch-findbugs-hadoop-hdds_container-service.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/5/artifact/out/patch-unit-hadoop-hdds_common.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/5/artifact/out/patch-unit-hadoop-hdds_container-service.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/5/testReport/ |
| Max. process+thread count | 446 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-hdds/container-service U: hadoop-hdds |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/5/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 26 | Docker mode activated. |
||| _ Prechecks _ |
| 0 | yamllint | 0 | yamllint was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 5 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for branch |
| +1 | mvninstall | 1041 | trunk passed |
| +1 | compile | 71 | trunk passed |
| +1 | checkstyle | 29 | trunk passed |
| -1 | mvnsite | 28 | container-service in trunk failed. |
| +1 | shadedclient | 727 | branch has no errors when building and testing our client artifacts. |
| -1 | findbugs | 24 | container-service in trunk failed. |
| +1 | javadoc | 63 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 10 | Maven dependency ordering for patch |
| -1 | mvninstall | 35 | container-service in the patch failed. |
| +1 | compile | 68 | the patch passed |
| +1 | javac | 68 | the patch passed |
| -0 | checkstyle | 24 | hadoop-hdds: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 63 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 3 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 728 | patch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 131 | the patch passed |
| +1 | javadoc | 58 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 63 | common in the patch failed. |
| -1 | unit | 70 | container-service in the patch failed. |
| +1 | asflicense | 24 | The patch does not generate ASF License warnings. |
| | | 3484 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.net.TestNodeSchemaManager |
|   | hadoop.hdds.scm.net.TestNetworkTopologyImpl |
|   | hadoop.ozone.container.common.TestDatanodeStateMachine |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-661/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/661 |
| JIRA Issue | HDDS-976 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  xml  findbugs  checkstyle  yamllint  |
| uname | Linux 4654c6113c21 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / d2637cb |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/6/artifact/out/branch-mvnsite-hadoop-hdds_container-service.txt |
| findbugs | v3.1.0-RC1 |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/6/artifact/out/branch-findbugs-hadoop-hdds_container-service.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/6/artifact/out/patch-mvninstall-hadoop-hdds_container-service.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/6/artifact/out/diff-checkstyle-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/6/artifact/out/patch-unit-hadoop-hdds_common.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/6/artifact/out/patch-unit-hadoop-hdds_container-service.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/6/testReport/ |
| Max. process+thread count | 453 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-hdds/container-service U: hadoop-hdds |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/6/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 25 | Docker mode activated. |
||| _ Prechecks _ |
| 0 | yamllint | 0 | yamllint was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 5 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 12 | Maven dependency ordering for branch |
| +1 | mvninstall | 989 | trunk passed |
| +1 | compile | 69 | trunk passed |
| +1 | checkstyle | 27 | trunk passed |
| +1 | mvnsite | 75 | trunk passed |
| +1 | shadedclient | 749 | branch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 122 | trunk passed |
| +1 | javadoc | 61 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 12 | Maven dependency ordering for patch |
| +1 | mvninstall | 75 | the patch passed |
| +1 | compile | 74 | the patch passed |
| +1 | javac | 74 | the patch passed |
| +1 | checkstyle | 23 | the patch passed |
| +1 | mvnsite | 65 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 3 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 732 | patch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 132 | the patch passed |
| +1 | javadoc | 57 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 65 | common in the patch failed. |
| -1 | unit | 72 | container-service in the patch failed. |
| +1 | asflicense | 25 | The patch does not generate ASF License warnings. |
| | | 3501 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.net.TestNetworkTopologyImpl |
|   | hadoop.hdds.scm.net.TestNodeSchemaManager |
|   | hadoop.ozone.container.common.TestDatanodeStateMachine |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-661/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/661 |
| JIRA Issue | HDDS-976 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  xml  findbugs  checkstyle  yamllint  |
| uname | Linux 730a4c61891b 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 993f36e |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/8/artifact/out/patch-unit-hadoop-hdds_common.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/8/artifact/out/patch-unit-hadoop-hdds_container-service.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/8/testReport/ |
| Max. process+thread count | 446 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-hdds/container-service U: hadoop-hdds |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-661/8/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by xiaoyuyao** [comment]
Thanks @cjjnjust  for the update. Some of the unit test failures seem related, can you check/fix them?

**Comment by cjjnjust** [comment]
@xiaoyuyao , I checked failed tests, it looks like not related to PR, it seems all errors are caused by `java.lang.RuntimeException: Fail to load schema file:/testptch/hadoop/hadoop-hdds/common/target/test-classes/networkTopologyTestFiles/good.xml, error:null`.  while I didn't remove it in this pr.

**Comment by xiaoyuyao** [comment]
+1, I will merge/commit the patch shortly. I manually retested the failed tests and they don't repro on my box.

**Pull request merged**